src/librustdoc/html/render.rs

```diff
@@ -2506,6 +2507,11 @@ fn render_impl(w: &mut fmt::Formatter, cx: &Context, i: &Impl, link: AssocItemLi
                 if !is_static || render_static {
                     let id = derive_id(format!("{}.{}", shortty, name));
                     write!(w, "<h4 id='{}' class='{}'>", id, shortty)?;
+                    let assoc_link = &AssocItemLink::Anchor(Some(id));
+                    let link = match link {
+                        &AssocItemLink::Anchor(None) => assoc_link,
```

Comment:
This is fine, but since you asked for suggestions... :smile:

You can write that as something like `if let AssocItemLink::Anchor(ref mut id) { *id = id }` (changing the function parameter to `mut link: AssocItemLink`).

You can also remove these pesky `&...` from match arms by doing `match *foo`.